snapcraft/plugins/_python/_pip.py

```diff
@@ -0,0 +1,263 @@
+        os.makedirs(self._python_package_dir, exist_ok=True)
+
+        self._python_command = get_python_command(
+            self._python_major_version, stage_dir=self._stage_dir,
+            install_dir=self._install_dir)
+
+        self._setup()
+
+    def _setup(self):
+        # Check to see if we have our own pip, yet. If not, we need to use the
+        # pip on the host (installed via build-packages) to grab our own.
+        try:
+            self._run([], stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode(sys.getfilesystemencoding()).strip()
+            if 'no module named pip' in output.lower():
+                logger.info('Fetching pip, setuptools, and wheel...')
+                env = {'PYTHONHOME': '/usr'}
+
+                # Using the host's pip, install our own pip and other tools we
+                # need.
+                self.download({'pip', 'setuptools', 'wheel'}, env=env)
+                self.install({'pip', 'setuptools', 'wheel'}, env=env)
+            else:
+                raise e
+
+    def download(self, args, **kwargs):
+        """Using the requested args, download packages into a cache.
+
+        Simple wrapper for `pip download`. Args are appended at the end, and
+        any keyword args are passed along to `common.run_output`. Note that
+        this function does not install the fetched packages.
+
+        :param list args: Arguments for `pip download`. Package names, for
+                          example, or the path to a dir containing a setup.py.
+        """
+        if not args:
+            return  # No packages to download
+
+        # Using pip with a few special parameters:
+        #
+        # --disable-pip-version-check: Don't whine if pip is out-of-date with
+        #                              the version on pypi.
+        # --dest: Download packages into the directory we've set aside for it.
+        self._run(['download', '--disable-pip-version-check', '--dest',
+                   self._python_package_dir] + list(args), **kwargs)
+
+    def install(self, args, **kwargs):
+        """Using the requested args, install packages from the cache.
+
+        Simple wrapper for `pip install`. Args are appended at the end, and
+        any keyword args are passed along to `common.run_output`. Note that
+        the packages should have already been downloaded via `self.download`.
+
+        :param list args: Arguments for `pip install`. Package names, for
+                          example, or the path to a dir containing a setup.py.
+        """
+        if not args:
+            return  # No packages to install
+
+        # Using pip with a few special parameters:
+        #
+        # --user: Install packages to PYTHONUSERBASE, which we've pointed to
+        #         the installdir.
+        # --no-compile: Don't compile .pyc files. FIXME: Not sure why
```

Comment:
Although this is a refactor, I wanted to make sure I didn't change any of the args we were using. However, I wanted to carefully document why we were using these args, and I didn't know the answer to some of them. @sergiusens, can you help me fill this out by answering my FIXMEs? There are some here in `install()` as well as below in `wheel()`.

Reply:
`--no-compile` is legacy and I think we should just remove it, it avoids needed to use `stage` and `prime` when using two parts, we even have a bug to remove it, so I say we just don't use it in this refactor.

Reply:
Okay, I'll drop it and give it a test.

Reply:
We agreed on chat that it makes more sense to leave this alone until the refactor has been validated, so I updated the FIXME to indicate such.